educator/src/Dscp/Educator/Web/Types.hs

```diff
@@ -113,6 +115,52 @@ data BlkProofInfo = BlkProofInfo
     , bpiTxs             :: [PrivateTx]
     } deriving (Show, Eq, Generic)
 
+---------------------------------------------------------------------------
+-- Buildable instances
+---------------------------------------------------------------------------
+
+instance Buildable (IsFinal) where
+    build (IsFinal{..}) =
+      "{ is final = " +| unIsFinal |+
+      " }"
+
+instance Buildable (HasProof) where
+    build (HasProof{..}) =
+      "{ has proof = " +| unHasProof |+
+      " }"
+
+instance Buildable (StudentInfo) where
+    build (StudentInfo{..}) =
+      "{ address = " +| siAddr |+
+      " }"
+
+instance Buildable (GradeInfo) where
+    build (GradeInfo{..}) =
+      "{ submission hash = " +| giSubmissionHash |+
+      ", grade = " +|| giGrade ||+
+      " }"
+
+instance Buildable (BlkProofInfo) where
+    build (BlkProofInfo{..}) =
+      "{ merkle tree = " +|| bpiMtreeSerialized ||+
```

Comment:
The whole merkle tree might be huge. I think that you should better display only root hash.